### PR TITLE
Syncronization testing after group deletingv1

### PR DESCRIPTION
**Pull request by pro-akim**
|Related issue|
|-------------|
|     #3953         |

## Description
Wazuh manager handles agent groups. 
If a group is deleted from a master cluster, there will be an instance where the agents require resynchronization (syncreq). 
If the group is deleted from a worker cluster, the cluster master will take care of reestablishing the group structure without the need for resynchronization.
This test suite tests the correct functioning of the mentioned use case.

Added
-test_group_synnc delete_folder_master
-test_group_synnc delete_folder_worker1
-test_group_synnc delete_folder_worker2

---

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @pro-akim  (Developer)  |           | ⚫⚫⚫ | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/11261167/report1.zip)[:green_circle:](https://github.com/wazuh/wazuh-qa/files/11261168/report2.zip)[:green_circle: ](https://github.com/wazuh/wazuh-qa/files/11261169/report3.zip)|         |         | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ |  ⚫⚫⚫  |        |         | Nothing to highlight |

### CODE REVIEW

File: tests/system/test_cluster/test_agent_groups/test_group_sync_status.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+inventory_path = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))),
+                              'provisioning', 'enrollment_cluster', 'inventory.yml')
+host_manager = HostManager(inventory_path)
+local_path = os.path.dirname(os.path.abspath(__file__))
+test_cases_yaml = read_yaml(os.path.join(local_path, 'data/test_group_sync_cases.yml'))
+wdb_query = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'script/wdb-query.py')
+
+@pytest.fixture
+def delete_logs():
```
Comment (juliamagan) [inline tests/system/test_cluster/test_agent_groups/test_group_sync_status.py:56]: Check if we can use `clean_environment()`

File: tests/system/test_cluster/test_agent_groups/test_group_sync_status.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        for agent_id in agent_ids:
+            response = host_manager.make_api_call(host=testinfra_hosts[0], token=host_manager.get_api_token(testinfra_hosts[0]), method='PUT',
+                                                      endpoint=f'/agents/{agent_id}/group/{group}')
+            assert response['status'] == 200, f"Failed to assign agent {agent_id} in {group} group: {response}"
+    
+@pytest.fixture
+def delete_group_folder(test_case):
+    groups_created = host_manager.run_command(testinfra_hosts[0], f'{WAZUH_PATH}/bin/agent_groups')
+    if test_case['test_case']['group_deleted'] in groups_created:
+        host_manager.run_command(test_case['test_case']['host'], f"rm -r {WAZUH_PATH}/etc/shared/{test_case['test_case']['group_deleted']} -f")
```
Comment (juliamagan) [inline tests/system/test_cluster/test_agent_groups/test_group_sync_status.py:89]: Check if we can use `delete_agent_groups()`

File: tests/system/test_cluster/test_agent_groups/test_group_sync_status.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    clean_cluster_logs(testinfra_hosts, host_manager)
+
+def delete_all_groups():
+    global groups_created
+    groups_created = host_manager.run_command(testinfra_hosts[0], f'{WAZUH_PATH}/bin/agent_groups')
+    for group in groups:
+        if group in groups_created:
+            response = host_manager.make_api_call(host=testinfra_hosts[0], token=host_manager.get_api_token(testinfra_hosts[0]), method='DELETE',
+                                                      endpoint=f'/groups?groups_list={group}')
+            assert response['status'] == 200, f"Failed to delete {group} group: {response}"
```
Comment (juliamagan) [inline tests/system/test_cluster/test_agent_groups/test_group_sync_status.py:66]: Check if we can use `delete_agent_groups()`

File: tests/system/test_cluster/test_agent_groups/test_group_sync_status.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    
+@pytest.fixture
+def wdb_query_creator():
+    wdb = read_file(wdb_query)
+    host_manager.modify_file_content(host=testinfra_hosts[0], path=f'{WAZUH_PATH}/wdb-query.py',content=wdb)
+
+def query_database(): 
+    query = 'global sql select group_sync_status from agent;'
+    response= host_manager.run_command(testinfra_hosts[0], f'python3 {WAZUH_PATH}/wdb-query.py "{query}"')
+    return response
```
Comment (juliamagan) [inline tests/system/test_cluster/test_agent_groups/test_group_sync_status.py:99]: Check if we can use `execute_wdb_query()`

File: tests/system/test_cluster/test_agent_groups/test_group_sync_status.py [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+    query = 'global sql select group_sync_status from agent;'
+    response= host_manager.run_command(testinfra_hosts[0], f'python3 {WAZUH_PATH}/wdb-query.py "{query}"')
+    return response 
+
+@pytest.fixture
+def kill_initial_syncreq():
+    result = query_database()
+    while 'syncreq' in result:
+        time.sleep(1)
+        result = query_database()
```
Comment (juliamagan) [inline tests/system/test_cluster/test_agent_groups/test_group_sync_status.py:106]:
```suggestion
@pytest.fixture
def wait_end_initial_syncreq():
    result = query_database()
    while 'syncreq' in result:
        time.sleep(1)
        result = query_database()
```

File: tests/system/test_cluster/test_agent_groups/test_group_sync_status.py [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+            first_time_check = "syncreq"
+
+@pytest.fixture          
+def second_check():
+    time.sleep(10)
+    global second_time_check
+    second_time_check = "synced"
+    result = query_database()
+    if 'syncreq' in result:
+        second_time_check = "syncreq"
```
Comment (juliamagan) [inline tests/system/test_cluster/test_agent_groups/test_group_sync_status.py:126]: This should be part of the test

**Review by juliamagan — CHANGES_REQUESTED** [review]
- The test is for 4.4.0, but the branch is `master` (4.6.0 at the moment)
- Delete the script that is no longer used
- Fix the checks
- Files have been duplicated when changing directories

**Comment by pro-akim** [comment]
|Related issue|
|-------------|
|     #3953         |

## Description
Wazuh manager handles agent groups. 
If a group is deleted from a master cluster, there will be an instance where the agents require resynchronization (syncreq). 
If the group is deleted from a worker cluster, the cluster master will take care of reestablishing the group structure without the need for resynchronization.
This test suite tests the correct functioning of the mentioned use case.


### Updated

-test_group_synnc delete_folder_master
-test_group_synnc delete_folder_worker1
-test_group_synnc delete_folder_worker2



---

## Testing performed




| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @pro-akim  (Developer)  |           | ⚫⚫⚫ | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/11344993/report1.zip)[:green_circle:](https://github.com/wazuh/wazuh-qa/files/11344994/report2.zip)[:green_circle: ](https://github.com/wazuh/wazuh-qa/files/11345016/report3.zip)|         |         | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ |  ⚫⚫⚫  |        |         | Nothing to highlight |
[report3.zip](https://github.com/wazuh/wazuh-qa/files/11344995/report3.zip)

File: tests/system/test_cluster/test_agent_groups/test_group_sync_status.py (hunk 6 above)
Comment (pro-akim) [reply tests/system/test_cluster/test_agent_groups/test_group_sync_status.py]: Checked and corrected in https://github.com/wazuh/wazuh-qa/pull/4143

File: tests/system/test_cluster/test_agent_groups/test_group_sync_status.py (hunk 3 above)
Comment (pro-akim) [reply tests/system/test_cluster/test_agent_groups/test_group_sync_status.py]: Checked and corrected in https://github.com/wazuh/wazuh-qa/pull/4143

File: tests/system/test_cluster/test_agent_groups/test_group_sync_status.py (hunk 2 above)
Comment (pro-akim) [reply tests/system/test_cluster/test_agent_groups/test_group_sync_status.py:119]: Checked and corrected in https://github.com/wazuh/wazuh-qa/pull/4143

File: tests/system/test_cluster/test_agent_groups/test_group_sync_status.py (hunk 1 above)
Comment (pro-akim) [reply tests/system/test_cluster/test_agent_groups/test_group_sync_status.py]: Checked and corrected in https://github.com/wazuh/wazuh-qa/pull/4143

**Pull request closed without merging**